index.js

```diff
@@ -42,6 +37,19 @@ class Ora {
 		this.linesToClear = 0;
 	}
 
+	get spinner() {
+		return this._spinner;
+	}
+
+	set spinner(sp) {
+		this.frameIndex = 0;
+		this._spinner = typeof sp === 'object' ? sp : (process.platform === 'win32' ? cliSpinners.line : (cliSpinners[sp] || cliSpinners.dots)); // eslint-disable-line no-nested-ternary
```

Comment:
Instead of `(cliSpinners[sp] || cliSpinners.dots)`, I think we should strictly validate the spinner and throw if it doesn't exist in `cliSpinners[sp]`.